src/query/functions/src/aggregates/aggregate_scalar_state.rs

```diff
@@ -180,6 +179,76 @@ where
     }
 }
 
+// Rust does not support SFINAE,
+// so we impl an independent struct for `any` because the behavior of `any` is little different from `min` and `max`.
+#[derive(Serialize, Deserialize, Default)]
+pub struct ScalarStateForAny<S: Scalar> {
+    #[serde(bound(deserialize = "S: DeserializeOwned"))]
+    pub value: Option<S>,
+}
+
+impl<S> ScalarStateFunc<S> for ScalarStateForAny<S>
```

Comment:
Can we have `ScalarState` for any ?

Reply:
Yes, this was what I did for my first implementation. But `ScalarState.add_batch()` will iterate the whole column, which I think is not proper for `ANY` (since we only need the first non-null value). So I want to implement a specific specialization for `CmpAny`, like `impl<S> ScalarStateFunc<S> for ScalarState<S, CmpAny>`. But Rust does not support the SFINAE mechanism like C++, so we can't write such codes.

Looking for a better way.